src/main/java/org/testng/internal/TestInvoker.java

```diff
@@ -262,12 +262,13 @@ private String checkDependencies(ITestNGMethod testMethod, ITestNGMethod[] allTe
     // If this method depends on groups, collect all the methods that
     // belong to these groups and make sure they have been run successfully
     String[] groups = testMethod.getGroupsDependedUpon();
+    ITestNGMethod[] allTestMethods = context.getAllTestMethods();
     if (null != groups && groups.length > 0) {
       // Get all the methods that belong to the group depended upon
       for (String element : groups) {
         ITestNGMethod[] methods =
             MethodGroupsHelper.findMethodsThatBelongToGroup(
-                testMethod, m_testContext.getAllTestMethods(), element);
+                testMethod, allTestMethods, element);
```

Comment:
If `m_testContext` and `context` are always the same, you don't need any `context` param

Reply:
There's a bit of clean-up that is needed to be done with respect to `ITestContext`. So I would want to handle this as a separate PR.